### DISCUSSION TITLE
Week 6 Video, reading, technical challenges & plan

### DISCUSSION THREAD

**Discussion by kassyoveson** [USU-ITLS-x265-23-summer/class-repo#122] [General]
**Thoughts on Video and Reading Assignment**
This week has been a fun week. I liked watching videos about people with disabilities. It's a good reminder of how to treat others with disabilities. The Mozella firefox article was interesting. It is not something that is normally taught in a college class, but I really did enjoy reading it. There are many good quotes from it that I actually wrote down such as “we have a desire to sacrifice ourselves to the ones we love.” It was longer than I expected, but I enjoyed what I read. 

**Technical Challenges**
Everything will be technically challenging, just kidding. It will be interesting writing my own code from scratch and putting the elements into it. I am still trying to figure out how java works with html and CSS. I am guessing that we will be using a platform in GitHub. I am worried about finding the bugs when I make a mistake or not being able to fully understand others' code to be of any use to them. I guess that's why I am taking this class. 

**Mitigation Plans**
I am hoping to start on my final projects soon. In doing so, I am hoping to take out some of the stress before the final deadline. I will be reaching out to you guys if I get stuck, which will probably happen. My first step is to do a sketch with the elements that I want to include in my web page and then figuring out how to code it to look and say what I need it to say. Is there a min or max of how long our code should be for this project? How complicated does it need to be?

**Comment by metasean** [comment]
> Everything will be technically challenging, just kidding.

Thank you for the laugh, it was much needed!

fyi: `java !== JavaScript` (sorta like `car !== carpet`)

> guessing that we will be using a platform in GitHub

You guess correctly.

> I am worried about finding the bugs when I make a mistake

You'll be covering that shortly.  You also might want to ask @Skye404 if she has any advice 😉 

> not being able to fully understand others' code to be of any use to them

It can definitely be hard at first, but doing so will also help you level up. 💯❗

> Is there a min or max of how long our code should be for this project? How complicated does it need to be? 

Review the [Project details](https://github.com/USU-ITLS-x265-23-summer/class-repo/blob/6d0b888462cc24b25722c5866a01bf7d76005702/Weekly-assignments/Projects.md), paying special attention to the "Project Assessment" section.

**Comment by Skye404** [comment]
If you go to Resources then Notes in the class repo, I've added this there for reference:

Tips for Debugging: 
1) Add console.log(*with your variable here*) at the end of your code. This is super helpful!!
2) Copy code to this page: https://pythontutor.com/javascript.html#mode=edit and click "Visualize Execution."
